### ISSUE TITLE
Style sheet being ignored

### ISSUE THREAD

**Issue by julesbl** [dompdf/dompdf#3455]
Hi

Using version 2.0.4 with php 7.4

I've got a weird thing happening I'm pretty sure I've set the chroot correctly but the style sheet specified in my document is still not loading.

`
$this->dompdf->getOptions()->setChroot(APPPATH. 'Views/reports/dompdf_assets/');
`

The directory is not in a public area.

the html is as follows

`<!DOCTYPE html>
<html lang="en-GB">
	<head>
		<title>Testing report</title>
		<style href="{{ site_url('general_report.css') }}"></style>
	</head>
	<body>
		{% include 'reports/header.twig.html' %}
		{% include 'reports/footer.twig.html' %}
		<p>Report content</p>
	</body>
</html>`

Style sheet is

`@page {
    margin: 100px 25px;
}

header {
    position: fixed;
    top: -60px;
    left: 0px;
    right: 0px;
    height: 50px;

    /** Extra personal styles **/
    background-color: #03a9f4;
    color: white;
    text-align: center;
    line-height: 35px;
}

footer {
    position: fixed;
    bottom: -60px;
    left: 0px;
    right: 0px;
    height: 50px;

    /** Extra personal styles **/
    background-color: #03a9f4;
    color: white;
    text-align: center;
    line-height: 35px;
}`

I've not tried loading any images yet

Jules

**Comment by bsweeney** [comment]
The stylesheet is referenced using a relative path so it will be interpreted based on either:
1. the location of the HTML document if you're loading a file using loadHtmlFile
2. the location of the executing PHP script if you're loading a string containing the HTML using loadHtml

**Comment by julesbl** [comment]
So in this case, I'm using a html string, I have to supply a path relative to index.php?

app->Views->reports->dompdf_assets->general_report.css
public->index.php

`<style` href="../app/Views/reports/dompdf_assets/general_report.css"></style>

this should work?

**Comment by julesbl** [comment]
Just sorted out an inline hack to get the style sheets in, not the best solution but it will have to do

**Comment by bsweeney** [comment]
I would expect the path to work, have you confirmed that APPPATH ends with a slash?

**Comment by julesbl** [comment]
Hi, 
Yes I did, first thing I did was check the path.

I've tried a number of things and attempted to check everything, probably can't see it for looking

**Comment by bsweeney** [comment]
Happens to me all the time. You could try [enabling additional debug messaging](https://github.com/dompdf/dompdf/wiki/Troubleshooting-Dompdf#configuration) to see if it gives you enough info to determine what's going on.

**Comment by julesbl** [comment]
Yes, tried all that, was like the style sheet request did not even exist, I
can usually fix these sorts of things if I can get an error

**Comment by bsweeney** [comment]
Is

```
<style href="../app/Views/reports/dompdf_assets/general_report.css"></style>
```

the code you're using? It should be something along the lines of

```
<link rel="stylesheet" href="./app/Views/reports/dompdf_assets/general_report.css">
```
